frontend/src/app/download/download.component.ts

```diff
@@ -142,11 +146,40 @@ export class DownloadComponent implements OnChanges {
             });
     }
 
-    private getCsvFields(): CorpusField[] {
+    private enableDirectDownload(totalResults: number) {
```

Comment:
Missing a return type, though obvious from the code. Could also omit the parameter and use `this.totalResults`?